<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BFSViewClass</class>
 <widget class="QWidget" name="BFSViewClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1098</width>
    <height>527</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QLabel" name="bfsInfoConsoleLabel">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>310</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Theory:</string>
   </property>
  </widget>
  <widget class="QLabel" name="bfsConsoleLabel">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>110</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Console:</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="bfsInfoTextBrowser">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>330</y>
     <width>371</width>
     <height>191</height>
    </rect>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#ffffff;&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; font-weight:600; color:#202122;&quot;&gt;Breadth-first search&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; (&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; font-weight:600; color:#202122;&quot;&gt;BFS&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt;) is an &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Algorithm&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;algorithm&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; for searching a &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Tree_(data_structure)&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;tree&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; data structure for a node that satisfies a given property. It starts at the &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Tree_(data_structure)#Terminology&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;tree root&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; and explores all nodes at the present &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Tree_(data_structure)#Terminology&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;depth&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; prior to moving on to the nodes at the next depth level. Extra memory, usually a &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Queue_(data_structure)&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;queue&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt;, is needed to keep track of the child nodes that were encountered but not yet explored.&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#ffffff;&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt;For example, in a &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Chess_endgame&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;chess endgame&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; a &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Chess_engine&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;chess engine&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; may build the &lt;/span&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Game_tree&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; text-decoration: underline; color:#0645ad;&quot;&gt;game tree&lt;/span&gt;&lt;/a&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; from the current position by applying all possible moves, and use breadth-first search to find a win position for white. Implicit trees (such as game trees or other problem-solving trees) may be of infinite size; breadth-first search is guaranteed to find a solution node if one exists.&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;  &lt;/span&gt;&lt;/p&gt;
&lt;h2 style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#ffffff;&quot;&gt;&lt;span style=&quot; font-family:'Georgia','serif'; font-size:8pt; font-weight:600; color:#000000;&quot;&gt;Pseudocode&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt; &lt;/span&gt;&lt;/h2&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#ffffff;&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; font-weight:600; color:#202122;&quot;&gt;Input&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt;: A graph &lt;/span&gt;&lt;span style=&quot; font-size:12.5pt; font-style:italic; color:#202122;&quot;&gt;G&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; and a &lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; font-style:italic; color:#202122;&quot;&gt;starting vertex&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-size:12.5pt; font-style:italic; color:#202122;&quot;&gt;root&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; of &lt;/span&gt;&lt;span style=&quot; font-size:12.5pt; font-style:italic; color:#202122;&quot;&gt;G&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#ffffff;&quot;&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; font-weight:600; color:#202122;&quot;&gt;Output&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt;: Goal state. The &lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; font-style:italic; color:#202122;&quot;&gt;parent&lt;/span&gt;&lt;span style=&quot; font-family:'Arial','sans-serif'; font-size:10.5pt; color:#202122;&quot;&gt; links trace the shortest path back to &lt;/span&gt;&lt;span style=&quot; font-size:12.5pt; font-style:italic; color:#202122;&quot;&gt;root&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 1  &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;procedure&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; BFS(&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;G&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;, &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;root&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;) &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;is&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 2      let &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;Q&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; be a queue&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 3      label &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;root&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; as explored&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 4      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;Q&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;.enqueue(&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;root&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 5      &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;while&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;Q&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; is not empty &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;do&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 6          &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;v&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; := &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;Q&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;.dequeue()&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 7          &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;if&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;v&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; is the goal &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;then&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 8              &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;return&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;v&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; 9          &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;for all&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; edges from &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;v&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; to &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;w&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;in&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;G&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;.adjacentEdges(&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;v&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;) &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;do&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;10              &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;if&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;w&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; is not labeled as explored &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-weight:600; color:#000000;&quot;&gt;then&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;11                  label &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;w&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt; as explored&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;12                  &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;w&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;.parent := &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;v&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; background-color:#f8f9fa;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;13                  &lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;Q&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;.enqueue(&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; font-style:italic; color:#000000;&quot;&gt;w&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:10.5pt; color:#000000;&quot;&gt;)&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:8pt;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;  &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; font-weight:600; color:#232629; background-color:#ffffff;&quot;&gt;Time complexity&lt;/span&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; color:#232629; background-color:#ffffff;&quot;&gt; is &lt;/span&gt;&lt;span style=&quot; font-family:'var(--ff-mono)','serif'; font-size:10pt; color:#232629;&quot;&gt;O(|V|)&lt;/span&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; color:#232629; background-color:#ffffff;&quot;&gt;, where &lt;/span&gt;&lt;span style=&quot; font-family:'var(--ff-mono)','serif'; font-size:10pt; color:#232629;&quot;&gt;|V|&lt;/span&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; color:#232629; background-color:#ffffff;&quot;&gt; is the number of nodes. You need to traverse all nodes.&lt;/span&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; color:#232629;&quot;&gt;&lt;br /&gt;&lt;/span&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; font-weight:600; color:#232629; background-color:#ffffff;&quot;&gt;Space complexity&lt;/span&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; color:#232629; background-color:#ffffff;&quot;&gt; is &lt;/span&gt;&lt;span style=&quot; font-family:'var(--ff-mono)','serif'; font-size:10pt; color:#232629;&quot;&gt;O(|V|)&lt;/span&gt;&lt;span style=&quot; font-family:'Segoe UI','sans-serif'; font-size:11.5pt; color:#232629; background-color:#ffffff;&quot;&gt; as well - since at worst case you need to hold all vertices in the queue&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;  &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;                     &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QWidget" name="bfsAnimAreaPane" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>691</width>
     <height>411</height>
    </rect>
   </property>
   <widget class="QGraphicsView" name="bfsAnimArea">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>701</width>
      <height>411</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="Line" name="bfsAnimTextAreaSeparator">
   <property name="geometry">
    <rect>
     <x>695</x>
     <y>110</y>
     <width>31</width>
     <height>411</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="bfsPlainTextEdit">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>130</y>
     <width>371</width>
     <height>171</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>691</width>
     <height>97</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="bfsHeaderAndControlPane">
    <item>
     <widget class="QLabel" name="bfsHeaderLabel">
      <property name="layoutDirection">
       <enum>Qt::RightToLeft</enum>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt; font-weight:600;&quot;&gt;BREADTH FIRST SEARCH&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="bfsControlPane">
      <item>
       <widget class="QLineEdit" name="bfsInsertLineEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="bfsInsertBtn">
        <property name="text">
         <string>Run</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="bfsInsertNewGraphSepLine">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bfsNewGraphBtn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>New Graph</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="bfsHomeBtn">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>10</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Home</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
